app/graphql/types/edition_type.rb

```diff
@@ -43,6 +43,22 @@ def not_stored_in_publishing_api
     alias_method :publishing_scheduled_at, :not_stored_in_publishing_api
     alias_method :scheduled_publishing_delay_seconds, :not_stored_in_publishing_api
 
+    class Translation < Types::BaseObject
+      field :locale, String
+      field :base_path, String
+    end
+
+    class EditionLinks < Types::BaseObject
+      field :available_translations, [Translation]
+
+      def available_translations
+        Presenters::Queries::AvailableTranslations.by_edition(object)
+          .translations.fetch(:available_translations, [])
+      end
```

Comment:
**thought:** genericness

Is this link type available/present on all editions/types? If not, I've wondered before about a mixin approach for shared links that aren't quite generic enough to appear everywhere (e.g. `include AvailableTranslationsLinks` in the relevant type classes). This feels related to [the conversation over on the other PR](https://github.com/alphagov/publishing-api/pull/3024#discussion_r1871426511)!

Similar thoughts on the next commit - where/if to draw the line between generic and specific/how precise to be with our typing